src/rust/ide/json-rpc/src/macros.rs

```diff
@@ -112,15 +112,18 @@ macro_rules! make_rpc_methods {
         /// Mock used for tests.
         #[derive(Debug,Default)]
         pub struct MockClient {
+            expect_all_calls : Cell<bool>,
             $($method_result : RefCell<HashMap<($($param_ty),*),Result<$result>>>,)*
         }
 
         impl API for MockClient {
             $(fn $method(&self $(,$param_name:$param_ty)*)
             -> std::pin::Pin<Box<dyn Future<Output=Result<$result>>>> {
-                let mut result = self.$method_result.borrow_mut();
-                let result     = result.remove(&($($param_name),*)).unwrap();
-                Box::pin(async move { result })
+                let mut results = self.$method_result.borrow_mut();
+                let params      = ($($param_name),*);
+                let result      = results.remove(&params);
```

Comment:
Actually this remark is to the cleint — but should we remove entries from the map? 
With such we have no means to handle two repeated calls with the same arguments.

I'd either not remove (but that collides with this `expect_all_calls`) or use some kind of MultiMap